r/R/dataset-scan.R

```diff
@@ -28,7 +28,7 @@
 #'
 #' * `dataset`: A `Dataset` or `arrow_dplyr_query` object, as returned by the
 #'    `dplyr` methods on `Dataset`.
-#' * `projection`: A character vector of column names to select
+#' * `projection`: A character vector of column names to select columns
```

Comment:
As it turns out, `projection` can be a named list of Expressions, just not when dataset is an arrow_dplyr_query (and thus already contains a named list of Expressions)